doc/user/content/sql/explain-timestamp.md

```diff
@@ -0,0 +1,192 @@
+    pub timestamp_context: TimestampContext<T>,
+    /// The largest timestamp not in advance of upper.
+    pub largest_not_in_advance_of_upper: T,
+}
+
+
+*Query timestamp: The timestamp in a timeline at which the query makes the read
+oracle read: The value of the timeline's oracle timestamp, if used.
+largest not in advance of upper: The largest timestamp not in advance of upper.
+upper: The write frontier of all involved sources.
+since: The read frontier of all involved sources.
+can respond immediately: True when the write frontier is greater than the query timestamp.
+timeline: The type of timeline the query's timestamp belongs:
+      /// EpochMilliseconds means the timestamp is the number of milliseconds since
+      /// the Unix epoch.
+      EpochMilliseconds,
+      /// External means the timestamp comes from an external data source and we
+      /// don't know what the number means. The attached String is the source's name,
+      /// which will result in different sources being incomparable.
+      External(String),
+      /// User means the user has manually specified a timeline. The attached
+      /// String is specified by the user, allowing them to decide sources that are
+      /// joinable.
+      User(String),
+
+Each source contains two frontiers:
+  Read: At which time
+  Write:
+
+                 query timestamp: 1673612424151 (2023-01-13 12:20:24.151) +
+           oracle read timestamp: 1673612424151 (2023-01-13 12:20:24.151) +
+ largest not in advance of upper: 1673612424151 (2023-01-13 12:20:24.151) +
+                           upper:[1673612424152 (2023-01-13 12:20:24.152)]+
+                           since:[1673612423000 (2023-01-13 12:20:23.000)]+
+
+
+                                 Timestamp
+---------------------------------------------------------------------------
+                 query timestamp: 1673612424151 (2023-01-13 12:20:24.151) +
+           oracle read timestamp: 1673612424151 (2023-01-13 12:20:24.151) +
+ largest not in advance of upper: 1673612424151 (2023-01-13 12:20:24.151) +
+                           upper:[1673612424152 (2023-01-13 12:20:24.152)]+
+                           since:[1673612423000 (2023-01-13 12:20:23.000)]+
+         can respond immediately: true                                    +
+                        timeline: Some(EpochMilliseconds)                 +
+                                                                          +
+ source materialize.public.a (u2014, storage):                            +
+                   read frontier:[1673612423000 (2023-01-13 12:20:23.000)]+
+                  write frontier:[1673612424152 (2023-01-13 12:20:24.152)]+ -->
+
+## Privileges
+
+The privileges required to execute this statement are:
+
+- `USAGE` privileges on the schemas that all relations and types in the query are contained in.
+- `USAGE` privileges on all types used in the query.
+- `SELECT` privileges on all relations in the query.
```

Comment:
It's probably worth adding this note here too

```
- NOTE: if any item is an index or a materialized view, then its owner must also have the necessary privileges to
    execute its definition.
```

Reply:
The `EXPLAIN TIMESTAMP` syntax is currently defined only for `SELECT` queries.

Reply:
Yes but this applies to following query for example (given that `mv` is a materialized view):

```SQL
EXPLAIN TIMESTAMP FOR SELECT * FROM mv;
```

Reply:
I mixed up the wording because `item` for me was more like the whole expeinee subject, I will change this to `relation`.